common/core/gpudevice.cpp

```diff
@@ -201,6 +201,8 @@ void GpuDevice::DisplayManager::HotPlugEventHandler() {
           "UpdateDisplayState.");
       UpdateDisplayState();
     }
+    /* Adding sleep to avoid busy looping before a retry */
+    usleep(1000);
```

Comment:
Kalyan, The loop is breaking / returning only when read fails. There could be a case of race / priority inversion (very corner case) where this thread will hog entire system, sleep ensures that wont happen. We should either remove the busy loop itself or have this sleep on the safer side. There is no perf improve due to this but this is more of precautionary measure. Busy loops are bad!